src/ToricVarieties/NormalToricVarieties/attributes.jl

```diff
@@ -317,6 +317,106 @@ end
 export toric_ideal
 
 
+@doc Markdown.doc"""
+    set_coordinate_names_of_torus(v::AbstractNormalToricVariety, coordinate_names::Vector{String})
+
+Allows to set the names of the coordinates of the torus.
+"""
+function set_coordinate_names_of_torus(v::AbstractNormalToricVariety, coordinate_names::Vector{String})
+    if length(coordinate_names) != ambient_dim(v)
+        throw(ArgumentError("The provided list of coordinate names must match the ambient dimension of the fan."))
+    end
+    set_attribute!(v, :coordinate_names_of_torus, coordinate_names)
+end
+export set_coordinate_names_of_torus
+
+
+@doc Markdown.doc"""
+    coordinate_names_of_torus(v::AbstractNormalToricVariety)
+
+This method returns the names of the coordinates of the torus of
+the normal toric variety `v`. The default is `x1,...,xn`.
+"""
+coordinate_names_of_torus(v::AbstractNormalToricVariety) = get_attribute!(v, :coordinate_names_of_torus, ["x$(i)" for i in 1:ambient_dim(v)])
```

Comment:
The drawback of using this variant of get_attribute is that with it the default value is always computed, even if it is not needed, leading to unnecessary allocations and overhead m. This is avoided by putting its computation into a function resp a `do` block

Reply:
Thank you for pointing this out. Does the following avoid this? (If yes, just changed.)

> @attr Vector{String} function coordinate_names_of_torus(v::AbstractNormalToricVariety)
>     return ["x$(i)" for i in 1:ambient_dim(v)]
> end

Reply:
indeed it does